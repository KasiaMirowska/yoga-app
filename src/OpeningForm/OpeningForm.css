.form-container {
    background-color:rgb(243, 240, 239); 
}

.opening-form {
    padding: 10px;    
}

.form__group {
  padding: 15px 0 0;
}
 .form__group1 {
    padding: 0;
 }
.form__field {
  width: 60%;
  font-family: inherit;
  border: 0;
  border-bottom: 2px solid gray;
  outline: 0;
  font-size: 1rem;
  color: grey;
  padding: 7px 0;
  background: transparent;  
  transition: border-color 0.2s;
  margin: 5px;
}

placeholder {
    color: transparent; 
}

.flow-form {
    display:flex;
    justify-content: space-between;
    margin: auto;
 
}

.flow-selection {
    display:flex;
    justify-content: space-between;
    margin: auto; 
   
}

.form-small-container {
    flex:3;
    display: inline;
}

.button-container {
    flex:1;
    display: inline;
    vertical-align: bottom;
}

@media (min-width: 800px) {
    .opening-form {
        padding: 20px;
        width: 65%;
        margin: auto;
        justify-content: space-around; 
        display: flex;
    }

    .flow-form {
        margin: 10px;
        flex: 1;
        padding: 5px;
        display: flex;
        margin-left: 5%;
    }

    .flow-selection {
        margin: 10px;
        flex:1;
        padding: 5px;
        display: flex;
        margin-left: 5%;
    }
   
    .form-small-container {
        flex:3;
    }
    
    .button-container {
        flex: 1;
    }
}
