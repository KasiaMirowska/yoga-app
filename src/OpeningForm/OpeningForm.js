import React from 'react';
import YogaContext from '../Context';
import APIFlowCalls from '../services/API_Flow_service';
import TokenService from '../services/token-service';
import config from '../config';
import './OpeningForm.css';

export default class OpeningForm extends React.Component {
    static contextType = YogaContext;
    state = {
        selection: null,
        error: null,
    }

    componentDidMount = () => {
        APIFlowCalls.getAllUserFlows()
            .then(data => {
                this.context.setFlowsList(data);
            })
            .catch(err => {
                this.context.setError(err);
            });
    }

    handleSubmit = (e) => {
        e.preventDefault();
        const { newFlowName } = e.target;
        const newFlow = {
            title: newFlowName.value,
        };

        const token = TokenService.hasAuthToken(config.TOKEN_KEY);
        if (!token) {
            this.props.history.push(`/login`);
        } else {
            APIFlowCalls.postNewFlow(newFlow)
                .then(data => {
                    newFlowName.value = '';
                    this.context.setCurrentFlow(data);
                    this.props.history.push('/flow');
                })
                .catch(res => {
                    console.log(res);
                    this.setState({
                        error: res,
                    });
                });
        }
    }

    onSelectFlow = (e) => {
        e.preventDefault();
        this.setState({
            selection: e.target.value
        });
    }

    enterFlow = () => {
        
        const flow = this.context.flows.find(flow => flow.title === this.state.selection);
       console.log('here')
        this.context.setCurrentFlow(flow)
        this.context.enterFlow(flow);
       
        this.props.history.push(`/flow`);
    }

    render() {
        const flowListName = this.context.flows.map((flow, key) => {
            return (
                <option key={key} value={flow.title}>{flow.title}</option>
            );
        });

        return (
            <div className='form-container'>
                <div className='error'>
                    {this.state.error ? this.state.error.message : null}
                </div>
                {(this.props.location.pathname === '/flowForm') ?
                    <div>
                        <h2>Please create a new flow or select an existing one.</h2>
                    </div>
                    : null}
                <div className='opening-form'>
                    <form onSubmit={this.handleSubmit} className='flow-form'>

                        <div className='form__group1 field form-small-container'>
                            <h3 className='flow-choice'>Create a new flow:</h3>
                            <input type="input" className="form__field" placeholder="flow name" name="name" id='newFlowName'  />
                        </div>
                        <div className='button-container'>
                            <button type='submit'>enter </button>
                        </div>
                    </form>

                    {this.context.flows.length > 0 ? <div className='flow-selection'>
                        <div className='form-small-container'>
                            <h3 className='flow-choice'>Choose exisiting flow: </h3>
                            <select className="form__field" onChange={this.onSelectFlow}>
                                <option value=''>Pick a flow</option>
                                {flowListName}
                            </select>
                        </div>
                        <div className='button-container'>
                            {this.state.selection === null ? <button disabled={!this.state.selection} className='disabled' onClick={this.enterFlow}> enter</button> : <button onClick={this.enterFlow}> enter</button>}
                        </div>
                    </div>
                        : null}
                </div>   
            </div>
        );
    }
}
