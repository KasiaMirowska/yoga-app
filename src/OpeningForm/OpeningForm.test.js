import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import OpeningForm from './OpeningForm';
import {BrowserRouter as BR } from 'react-router-dom';


describe('OpeningForm component', () => {
    const props = {
        location: {
            pathname: '/',
        }
    };

    it ('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDom.render(<BR><OpeningForm {...props}/></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const form = renderer.create(<BR><OpeningForm {...props}/></BR>);
        expect(form.toJSON()).toMatchSnapshot();
    });
});
