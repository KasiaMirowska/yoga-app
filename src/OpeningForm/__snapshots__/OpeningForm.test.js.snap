// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpeningForm component renders UI as expected 1`] = `
<div
  className="form-container"
>
  <div
    className="error"
  />
  <div
    className="opening-form"
  >
    <form
      className="flow-form"
      onSubmit={[Function]}
    >
      <div
        className="form__group1 field form-small-container"
      >
        <h3
          className="flow-choice"
        >
          Create a new flow:
        </h3>
        <input
          className="form__field"
          id="newFlowName"
          name="name"
          placeholder="flow name"
          type="input"
        />
      </div>
      <div
        className="button-container"
      >
        <button
          type="submit"
        >
          enter 
        </button>
      </div>
    </form>
  </div>
</div>
`;
